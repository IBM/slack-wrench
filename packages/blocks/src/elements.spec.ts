import {
  Button,
  Markdown,
  OptionObject,
  OverflowMenu,
  PlainText,
} from './elements';

describe('Slack Element widgets', () => {
  const text = 'Ravenclaw is obviously the best house.';
  const buttonId = 'ravenAgree';

  it('renders markdown object', () => {
    expect.assertions(1);
    expect(Markdown(text)).toMatchSnapshot();
  });

  it('renders a plaintext object', () => {
    expect.assertions(1);
    expect(PlainText(text)).toMatchSnapshot();
  });

  it('renders a plaintext object without emoji', () => {
    expect.assertions(1);
    expect(PlainText(text, false)).toMatchSnapshot();
  });

  it('renders an option object', () => {
    expect.assertions(1);
    expect(OptionObject('Why best?', 'why-best-raven')).toMatchSnapshot();
  });

  it('renders a button element', () => {
    expect.assertions(1);
    expect(Button(text, buttonId)).toMatchSnapshot();
  });

  it('renders an overflow menu element', () => {
    expect.assertions(1);
    expect(
      OverflowMenu(
        [
          OptionObject('Why best?', 'why-best-raven'),
          OptionObject('Ping JK', 'notify-author'),
        ],
        'house-actions',
      ),
    ).toMatchSnapshot();
  });
});
