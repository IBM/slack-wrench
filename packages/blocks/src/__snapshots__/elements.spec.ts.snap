// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slack Element widgets renders a button element 1`] = `
Object {
  "action_id": "ravenAgree",
  "text": Object {
    "emoji": true,
    "text": "Ravenclaw is obviously the best house.",
    "type": "plain_text",
  },
  "type": "button",
}
`;

exports[`Slack Element widgets renders a plaintext object 1`] = `
Object {
  "emoji": true,
  "text": "Ravenclaw is obviously the best house.",
  "type": "plain_text",
}
`;

exports[`Slack Element widgets renders a plaintext object without emoji 1`] = `
Object {
  "emoji": false,
  "text": "Ravenclaw is obviously the best house.",
  "type": "plain_text",
}
`;

exports[`Slack Element widgets renders an option object 1`] = `
Object {
  "text": Object {
    "emoji": true,
    "text": "Why best?",
    "type": "plain_text",
  },
  "value": "why-best-raven",
}
`;

exports[`Slack Element widgets renders an overflow menu element 1`] = `
Object {
  "action_id": "house-actions",
  "options": Array [
    Object {
      "text": Object {
        "emoji": true,
        "text": "Why best?",
        "type": "plain_text",
      },
      "value": "why-best-raven",
    },
    Object {
      "text": Object {
        "emoji": true,
        "text": "Ping JK",
        "type": "plain_text",
      },
      "value": "notify-author",
    },
  ],
  "type": "overflow",
}
`;

exports[`Slack Element widgets renders markdown object 1`] = `
Object {
  "text": "Ravenclaw is obviously the best house.",
  "type": "mrkdwn",
}
`;
